addon/components/attach-popover.js

```diff
@@ -133,33 +112,63 @@ export default class AttachPopover extends Component {
     return hideOn === null ? [] : hideOn.split(' ');
   }
 
-  @computed('arrow', 'flip', 'modifiers')
-  get _modifiers() {
-    // Copy the modifiers since we might write to the provided hash
-    const modifiers
-      = this.modifiers ? Object.assign({}, this.modifiers) : {};
+  // eslint-disable-next-line ember/require-computed-property-dependencies
+  @computed('arrow', 'flip', 'middleware')
+  get _middleware() {
+    // Copy the middleware since we might write to the provided array
+    const middleware
+      = this.middleware ? [...this.middleware] : [];
 
-    const arrow = this.arrow;
-    if (typeof(arrow) === 'boolean' && !modifiers.arrow) {
-      modifiers.arrow = { enabled: arrow };
+    if (!!this.arrow && !middleware.find(name => name === 'arrow')) {
+      middleware.push(arrow({ element: this._arrowElement }));
     }
 
     const flipString = this.flip;
     if (flipString) {
-      const flipModifier = { behavior: flipString.split(' ') };
-      if (!modifiers.flip) {
-        modifiers.flip = flipModifier;
-      } else if (!modifiers.flip.behavior) {
-        // Copy the flip modifier since we are altering the provided hash
-        modifiers.flip = Object.assign({}, modifiers.flip, flipModifier);
+      const flipOptions = { fallbackPlacements: flipString.split(' ') };
+      const flipMiddleware = middleware.find(name => name === 'flip');
+      if (!flipMiddleware) {
+        middleware.push(flip(flipOptions));
+      } else if (!flipMiddleware.fallbackPlacements) {
+        Object.assign({}, flipMiddleware, flipOptions);
       }
     }
 
-    return modifiers;
+    return middleware;
+  }
+
+  @computed('_parentFinder.parentNode', '_renderInPlace', 'floatingElementContainer')
+  get _floatingElementContainer() {
+    const maybeContainer = this.floatingElementContainer;
+    const renderInPlace = this._renderInPlace;
+    let floatingElementContainer;
+
+    if (renderInPlace) {
+      floatingElementContainer = this._parentFinder.parentNode;
+    } else if (maybeContainer instanceof Element) {
+      floatingElementContainer = maybeContainer;
+    } else if (typeof maybeContainer === 'string') {
+      const selector = maybeContainer;
+      const possibleContainers = self.document.querySelectorAll(selector);
+
+      assert(`floatingElementContainer selector "${selector}" found `
+        + `${possibleContainers.length} possible containers when there should be exactly 1`, possibleContainers.length === 1);
+
+      floatingElementContainer = possibleContainers[0];
+    }
+
+    return floatingElementContainer;
+  }
```

Comment:
This is the only part taken from the `ember-popper` to process the floating element's container logic